src/main/java/com/google/devtools/build/lib/sandbox/WindowsSandboxedSpawn.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.sandbox;
+
+import com.google.devtools.build.lib.exec.TreeDeleter;
+import com.google.devtools.build.lib.sandbox.SandboxHelpers.SandboxOutputs;
+import com.google.devtools.build.lib.vfs.FileStatus;
+import com.google.devtools.build.lib.vfs.FileSystemUtils;
+import com.google.devtools.build.lib.vfs.Path;
+import com.google.devtools.build.lib.vfs.PathFragment;
+import com.google.devtools.build.lib.vfs.Symlinks;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Creates an execRoot for a Spawn by copying input files.
+ */
+public class WindowsSandboxedSpawn extends AbstractContainerizingSandboxedSpawn {
```

Comment:
This does the same thing as `CopyingSandboxedSpawn` for now as I just want to get things to work first.

Reply:
> Do you mean it still copies instead of something else?

Yes

> What will you replace copying with?

I was thinking to collect input file paths from `copyFile` method below so that I can mark them as readable in windows-sandbox. To do that, this class will hold a reference to the commandlinebuilder, which now I think is a horrible way to do stuff.

> Thinking more, what is the purpose of a unique execroot? I suppose it (a) allows staging the declared inputs only, and (b) it isolates actions so they don't overwrite each other's intermediate outputs.
I understand (b) is useful, but it seems (a) isn't required for the detours-based sandbox.

You are right. (a) isn't required for the detours-based sandbox. In the final stage, the sandbox will just run in the common execroot, files will be marked readable or read-write explicitly. (b) is nice to have, but not in the scope for now.

I just found out about `SandboxHelpers.processInputFiles` which seems to be what I need, so I think this is what I am going to do:

- Delete `WindowsSandboxedSpawn.java` this class and use `CopyingSandboxedSpawn` for this PR.
- Figure out how to use ``SandboxHelpers.processInputFiles` and use that when building windows-sandbox command line.
- Create a `DummySandboxedSpawn` for windows-sandbox which does nothing.

Reply:
SGTM, thank you!